src/viam/sdk/CMakeLists.txt

```diff
@@ -159,6 +159,16 @@ target_link_libraries(viamsdk
 )
 
 
+# Set compile and link options based on arguments
```

Comment:
Content here looks fine, but can you move this block up above `target_sources`? I feel like setting compile flags is something that should happen early in the sequence of defining the target. So either before or after where we stick the special API include path in.